source/component/PasDoc_TagManager.pas

```diff
@@ -661,6 +661,23 @@ function IsMarkdownSpecialChar(const Description: string; Offset: Integer;
   If found, then it also sets OffsetEnd to position of found sign.}
 function FindMarkdownSpecialSign(const Description: string; Offset: Integer;
   const Sign: string; out OffsetEnd: Integer): boolean;
+
+  // Check if special sign in Description at Offset is escaped
+  // (has strictly single `\` before)
+  function IsEscaped(const Description: string; Offset: Integer): Boolean;
```

Comment:
What happens in case of 3 or 4 or more consecutive backslashes? E.g.

```
\\\\\\
```

The proper result should indicate that backslashes on 1, 3, 5 index are used only for escaping, while backslashes on positions 2, 4, 6 are to be displayed.

I don't think you can construct a reliable function `IsEcaped` like this :) The usual solution is to track "whether the last character was a backslash that should cause the next character to be escaped" during iteration over characters.

Reply:
You're right! I think I fixed this in commit below

Reply:
I don't think detecting using "odd number of backslashes" is correct either :)

Because you can have this:

```
\\ \\ \[ \\
```

So some backslashes are used to escape backslashes, some backslashes are used to escape `[`. You cannot use the `Odd(EscapesCount)` as reliable indicator to decide which backslash to remove.

Also, this would mean that each `IsEscaped` works in linear time, and you call `IsEscaped` in a loop, which probably can be called a ~linear number of times (with respect to input string length). So you would have something that can work in `O(n^2)` time.

I think the right way to do it is that `FindMarkdownSpecialSign` should search for the appropriate sign, iterate over characters, tracking whether "the last character is a backslash that is being used to escape". So you just have to modify `FindMarkdownSpecialSign` logic, to not use `PosEx` but use your own loop that looks at each consecutive character and tracks backslashes.

Reply:
Ah damn. I forgot to add some logic. I meant "loop until first non-escape char" of course. Amended the fix to the same commit.